### PR TITLE
simplify delegated operation doc

### PR DESCRIPTION
**Pull request by ritch**
- removes redundant `params` from operations docs
 - removes panel_state from operations docs, since it is unused

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method to enhance the serialization of request parameters by excluding specific keys.

- **Improvements**
	- Updated the constructor to enforce stricter type requirements for the context parameter.
	- Streamlined serialization and deserialization processes for the context attribute.
	- Enhanced error handling and data integrity within repository methods related to operation queuing and state updates.
	- Improved validation for progress parameters in the update progress method.

### CODE REVIEW

File: fiftyone/factory/repos/delegated_operation.py [hunk 1]
```diff
@@ -214,6 +214,7 @@ def queue_operation(self, **kwargs: Any) -> DelegatedOperationDocument:
             context.request_params["dataset_id"] = str(op.dataset_id)
             context.request_params["dataset_name"] = context.dataset.name
 
+        op.context = context
```
Comment (ritch) [inline fiftyone/factory/repos/delegated_operation.py:217]: This fixes an existing issue that I think might explain some of the weirdness/handling of `op.context` as `dict`!

Without this change [this](https://github.com/voxel51/fiftyone/pull/5291/files#diff-8c618b37055d71658b7666c9e677d916dfbb8b9f43876a5a2a983be19b63c100R192) is a no-op.

File: fiftyone/operators/executor.py [hunk 2]
```diff
@@ -889,6 +889,11 @@ def store(self, store_name):
         dataset_id = self.dataset._doc.id
         return ExecutionStore.create(store_name, dataset_id)
 
+    def _get_serialized_request_params(self):
```
Comment (ritch) [inline fiftyone/operators/executor.py:892]: Not really liking this as the interface...but didn't feel that changing `.serialize()` was the right move. However there is no need for `panel_state` in a serialized `ctx`. Instead it should really be outside of `request_params` to begin with. I've started that change elsewhere, but its a much more complex change and regresses some functionality in the dashboard plugin.

File: fiftyone/factory/repos/delegated_operation_doc.py [hunk 3]
```diff
@@ -24,17 +24,14 @@ def __init__(
...
     ):
         self.operator = operator
         self.label = None
         self.delegation_target = delegation_target
-        self.context = (
-            context.to_dict()
-            if isinstance(context, ExecutionContext)
-            else context
-        )
+        assert isinstance(context, ExecutionContext) or context is None
```
Comment (kaixi-wang) [inline fiftyone/factory/repos/delegated_operation_doc.py:33]: don't use assert

Comment (ritch) [reply fiftyone/factory/repos/delegated_operation_doc.py]: used this instead:

```python
        try:
            assert isinstance(context, ExecutionContext) or context is None
        except:
            raise AttributeError
```

File: fiftyone/factory/repos/delegated_operation_doc.py [hunk 4]
```diff
@@ -24,17 +24,17 @@ def __init__(
...
         self.operator = operator
         self.label = None
         self.delegation_target = delegation_target
-        self.context = (
-            context.to_dict()
-            if isinstance(context, ExecutionContext)
-            else context
-        )
+        try:
+            assert isinstance(context, ExecutionContext) or context is None
```
Comment (kaixi-wang) [inline fiftyone/factory/repos/delegated_operation_doc.py:34]: You can just do if... else raise.

**Review by kaixi-wang — APPROVED** [review]
Lgtm

**Review by CamronStaley — APPROVED** [review]

**Pull request merged**